### DISCUSSION TITLE
React 18 support

### DISCUSSION THREAD

**Discussion by jmvtrinidad** [testing-library/user-event#945] [Q&A]
Did anyone try this with react 18? it looks like it is not working properly, for example the click, fireEvent.click works fine but not userEvent.click.

**Comment by ph-fritsche** [comment]
Yes, I'm using the library with `react@18` without any problems.
We also have some tests for react-specific aspects which we run with `react@18` and `react@17`: https://github.com/testing-library/user-event/tree/main/tests/react 
Comparing to `fireEvent` is moot, as the APIs follow a different approach. See https://testing-library.com/docs/user-event/intro/#difference-to-fireevent

If you happen to have tests working with `react@17` but not with `react@18`, please try to create a minimal reproduction example and [file a bug report](https://github.com/testing-library/user-event/issues/new?assignees=&labels=needs+assessment%2Cbug&template=bug_report.yml) so that we can look into it.

**Comment by DrDr3ck** [comment]
remark, on my side, I now need to add await in front of userEvent.click to make it work in react18.

**Reply by Wxh16144** [reply]
I'm having a similar problem, and even putting await in front of it doesn't solve it. I finally solved it by using fireEvent.click
